pkg/apps/application.jsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React from "react";
+
+import PackageKit from "./packagekit.js";
+import { left_click, icon_url, show_error, launch, ProgressBar, CancelButton } from "./utils.jsx";
+
+import "./application.css";
+
+var _ = cockpit.gettext;
+
+class Application extends React.Component {
+    constructor() {
+        super();
+        this.state = { error: null, progress: false };
+    }
+
+    render() {
+        var self = this;
+        var state = this.state;
+        var metainfo_db = this.props.metainfo_db;
+        var comp;
+
+        if (!this.props.id)
+            return null;
+
+        comp = metainfo_db.components[this.props.id];
+
+        function action(func, arg, progress_title) {
+            self.setState({ progress_title: progress_title });
+            func(arg, data => self.setState({ progress: data })).
+                always(() => self.setState({ progress: false })).
+                fail(show_error);
+        }
+
+        function install() {
+            action(PackageKit.install, comp.pkgname, _("Installing"));
+        }
+
+        function remove() {
+            action(PackageKit.remove, comp.file, _("Removing"));
+        }
+
+        function render_description(desc) {
+            return desc.map(elt => {
```

Comment:
I think it wouldn't hurt to make these variable names, especially `elt`, a bit more verbose

Reply:
I have called it "paragraph".